sdk/python/kfp/components/executor_test.py

```diff
@@ -100,52 +43,189 @@ def setUp(cls):
+            return json.loads(f.read())
 
     def test_input_parameter(self):
+        executor_input = """\
+        {
+          "inputs": {
+            "parameterValues": {
+              "input_parameter": "Hello, KFP"
+            }
+          },
+          "outputs": {
+            "parameters": {
+              "output": {
+                "outputFile": "gs://some-bucket/output"
+              }
+            },
+            "outputFile": "%(test_dir)s/output_metadata.json"
+          }
+        }
+        """
 
-        def test_func(input_parameter: str):
+        def test_func(input_parameter: str) -> str:
             self.assertEqual(input_parameter, 'Hello, KFP')
+            return input_parameter
 
-        self._get_executor(test_func).execute()
+        output_metadata = self.execute_and_load_output_metadata(
+            test_func, executor_input)
+        self.assertEqual({'parameterValues': {
+            'Output': 'Hello, KFP'
+        }}, output_metadata)
+
+    def test_input_artifact_custom_type(self):
+        executor_input = """\
+        {
+          "inputs": {
+            "parameterValues": {
+            },
+            "artifacts": {
+              "input_artifact_one": {
+                "artifacts": [
+                  {
+                    "metadata": {},
+                    "name": "input_artifact_one",
+                    "type": {
+                      "schemaTitle": "google.VertexDataset"
+                    },
+                    "uri": "gs://some-bucket/input_artifact_one"
+                  }
+                ]
+              }
+            }
+          },
+          "outputs": {
+            "outputFile": "%(test_dir)s/output_metadata.json"
+          }
+        }
+        """
+
+        class VertexDataset:
+            schema_title = 'google.VertexDataset'
+            schema_version = '0.0.0'
+
+            def __init__(self, name: str, uri: str, metadata: dict) -> None:
+                self.name = name
+                self.uri = uri
+                self.metadata = metadata
+
+            @property
+            def path(self) -> str:
+                return self.uri.replace('gs://',
+                                        artifact_types._GCS_LOCAL_MOUNT_PREFIX)
+
+        def test_func(input_artifact_one: Input[VertexDataset]):
+            self.assertEqual(input_artifact_one.uri,
+                             'gs://some-bucket/input_artifact_one')
+            self.assertEqual(
+                input_artifact_one.path,
+                os.path.join(self._test_dir, 'some-bucket/input_artifact_one'))
+            self.assertEqual(input_artifact_one.name, 'input_artifact_one')
+            self.assertIsInstance(input_artifact_one, VertexDataset)
+
+        self.execute_and_load_output_metadata(test_func, executor_input)
 
     def test_input_artifact(self):
+        executor_input = """\
+        {
+          "inputs": {
+            "parameterValues": {
+            },
```

Comment:
If no input parameter, do we actually get this empty dict or we might not have `parameterValues` at all?
Maybe remove this and make sure we can still load executor input correctly.

Reply:
We do not have `parameterValues` at all